src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfoResolverWithAddedModifiers.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace System.Text.Json.Serialization.Metadata
+{
+    internal sealed class JsonTypeInfoResolverWithAddedModifiers : IJsonTypeInfoResolver
+    {
+        private readonly IJsonTypeInfoResolver _source;
+        private readonly Action<JsonTypeInfo>[] _modifiers;
```

Comment:
I'm curious if you could support it by List and count (you always append so technically you could reuse same instance) or ImmutableArray/List

Reply:
We want the resolver instances to be immutable -- if we reused the same `List<T>` then calling `var bar = foo.WithAddedModifiers(...);` then `foo` would be mutated.